ocaml/lexer.ml

```diff
@@ -30,6 +34,12 @@ type lexer = {
 let slurp fname = In_channel.with_file ~binary:false fname 
     ~f:(fun ch -> In_channel.input_all ch)
 
+let op_info op = 
```

Comment:
I would put this function in the parser since precedence is a parsing thing, not a lexing thing. This would also let you hide it entirely in the parser implementation and not have it exposed in any interface.